snapcraft/internal/cache/_snap.py

```diff
@@ -39,19 +40,47 @@ def cache(self, snap_filename, revision):
 
         :returns: path to cached revision.
         """
-        snap_cache_dir = self._setup_snap_cache()
-
         cached_snap = _rewrite_snap_filename_with_revision(
             snap_filename,
             revision)
-        cached_snap_path = os.path.join(snap_cache_dir, cached_snap)
+        cached_snap_path = os.path.join(self.snap_cache_dir, cached_snap)
         try:
             shutil.copyfile(snap_filename, cached_snap_path)
         except OSError:
             logger.warning(
                 'Unable to cache snap {}.'.format(cached_snap))
         return cached_snap_path
 
+    def prune(self, keep_revision, *args):
+        """Prune the snap revisions beside the keep_revision in XDG cache.
+
+        :returns: pruned file paths list.
+        """
+        if keep_revision is None:
```

Comment:
This check isn't needed, the one below does the job, if `None` is passed this is what an `int` conversion raises:

```python
int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```